### ISSUE TITLE
Consider picking up netkit scrubbing skb meta data kernel patch

### ISSUE THREAD

**Issue by jfroy** [siderolabs/pkgs#1104]
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?h=v6.13-rc1&id=d137af8768580378a6692c50db66c613ce9114cc

Without this patch, netkit scrubs the data that cilium relies on to classify flows as from/to cluster, defaulting those flows to world and thus potentially breaking connectivity in clusters using network policies.

**Comment by smira** [comment]
At this point I would rather disable netkit, as it seems to be broken at the moment?

**Comment by jfroy** [comment]
I don't know enough to make a general statement. The referenced issue is specifically about Cilium network policies. Connectivity without network policies does work (I have tested it some months ago). I can't speak to interactions with other features, or with other CNIs.

It's probably fine to keep it enabled in the kernel for experimentation, since at least Cilium requires opt-in (the default remains veth).

I am going to test a 6.12 kernel with the patch set applied soon. I am just waiting on openzfs to release for 6.12...

**Comment by smira** [comment]
I'm just afraid of the support issues saying that Cilium doesn't work (again).

Let's put this on hold until we see Linux 6.12.2 and whether the change is backported.

**Comment by smira** [comment]
It doesn't look like the change was backported to 6.12.2/.3

**Comment by smira** [comment]
But it got into 6.12.4: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.4

So we can consider this to be resolved as we update to 6.12.4.
